packages/agw-react/src/privy/usePrivyCrossAppProvider.ts

```diff
@@ -0,0 +1,210 @@
+    eth_getFilterChanges: true,
+    eth_getFilterLogs: true,
+    eth_getLogs: true,
+    eth_getProof: true,
+    eth_getStorageAt: true,
+    eth_getTransactionByBlockHashAndIndex: true,
+    eth_getTransactionByBlockNumberAndIndex: true,
+    eth_getTransactionByHash: true,
+    eth_getTransactionCount: true,
+    eth_getTransactionReceipt: true,
+    eth_getUncleByBlockHashAndIndex: true,
+    eth_getUncleByBlockNumberAndIndex: true,
+    eth_getUncleCountByBlockHash: true,
+    eth_getUncleCountByBlockNumber: true,
+    eth_maxPriorityFeePerGas: true,
+    eth_newBlockFilter: true,
+    eth_newFilter: true,
+    eth_newPendingTransactionFilter: true,
+    eth_protocolVersion: true,
+    eth_sendRawTransaction: true,
+    eth_uninstallFilter: true,
+  };
+  const passthrough = (method: EIP1474MethodNames) =>
+    !!passthroughMethods[method];
+
+  const publicClient = createPublicClient({
+    chain,
+    transport,
+  });
+
+  const getAddressFromUser = (user: User | null): Address | undefined => {
+    if (!user) {
+      return undefined;
+    }
+    const crossAppAccount = user.linkedAccounts.find(
+      (account) =>
+        account.type === 'cross_app' && account.providerApp.id === AGW_APP_ID,
+    ) as CrossAppAccount | undefined;
+
+    const address = crossAppAccount?.embeddedWallets?.[0]?.address;
+    return address ? (address as Address) : undefined;
+  };
+
+  const getAccounts = useCallback(
+    async (promptLogin: boolean) => {
+      if (!ready) {
+        return [];
+      }
+      let contextUser = user;
+      if (promptLogin) {
+        if (!contextUser && !authenticated) {
+          contextUser = await loginWithCrossAppAccount({
+            appId: AGW_APP_ID,
+          });
+        } else if (!contextUser && authenticated) {
+          contextUser = await linkCrossAppAccount({ appId: AGW_APP_ID });
+        }
+      }
+      const address = getAddressFromUser(contextUser);
+      return address ? [address] : [];
+    },
+    [user, authenticated, ready, loginWithCrossAppAccount, linkCrossAppAccount],
+  );
+
+  const eventListeners = new Map<string, ((...args: any[]) => void)[]>();
+
+  const handleRequest = useCallback(
+    async (request: any) => {
+      const { method, params } = request;
+      if (passthrough(method as EIP1474MethodNames)) {
+        return publicClient.request(request);
+      }
+
+      switch (method) {
+        case 'eth_requestAccounts': {
+          return await getAccounts(true);
+        }
+        case 'eth_accounts': {
+          return await getAccounts(false);
+        }
+        case 'wallet_switchEthereumChain':
+          // TODO: do we need to do anything here?
+          return null;
+        case 'wallet_revokePermissions':
+          // TODO: do we need to do anything here?
+          return null;
+        case 'eth_sendTransaction':
+        case 'eth_signTransaction':
+          // TODO: Implement
+          return randomBytes(32).toString('hex'); // fake tx hash
```

Comment:
What is the implementation going to be here?

Reply:
tbd - currently only `sendTransaction` is exposed via cross app connect, not a `signTransaction`

This only applies to transactions from the underlying cross connect EOA, not the AGW

Reply:
ah gotcha